modules/Microsoft.Network/networkManagers/.test/min/deploy.test.bicep

```diff
@@ -0,0 +1,49 @@
+targetScope = 'subscription'
+
+// ========== //
+// Parameters //
+// ========== //
+@description('Optional. The name of the resource group to deploy for testing purposes.')
+@maxLength(90)
+param resourceGroupName string = 'ms.network.networkmanagers-${serviceShort}-rg'
+
+@description('Optional. The location to deploy resources to.')
+param location string = deployment().location
+
+@description('Optional. A short identifier for the kind of deployment. Should be kept short to not run into resource-name length-constraints.')
+param serviceShort string = 'nnmmin'
+
+@description('Optional. Enable telemetry via a Globally Unique Identifier (GUID).')
+param enableDefaultTelemetry bool = true
+
+// =========== //
+// Deployments //
+// =========== //
+
+// General resources
+// =================
+resource resourceGroup 'Microsoft.Resources/resourceGroups@2021-04-01' = {
+  name: resourceGroupName
+  location: location
+}
+
+// ============== //
+// Test Execution //
+// ============== //
+
+module testDeployment '../../deploy.bicep' = {
+  scope: resourceGroup
+  name: '${uniqueString(deployment().name)}-test-${serviceShort}'
+  params: {
+    enableDefaultTelemetry: enableDefaultTelemetry
+    name: '<<namePrefix>>${serviceShort}001'
+    networkManagerScopeAccesses: [
+      'Connectivity'
+    ]
+    networkManagerScopes: {
+      managementGroups: [
+        '/providers/Microsoft.Management/managementGroups/<<managementGroupId>>'
```

Comment:
Can this be done via the `managementGroup()` function

Reply:
We'd need to change the scope to an MG scope deployment to use that function. Right?

Reply:
If so, nevermind 😄 will check later. You might be right that this function behaves a bit weird (for no reason)

Reply:
It is indeed based on the doco and what I've tested too:

See [Documentation](https://learn.microsoft.com/en-us/azure/azure-resource-manager/bicep/bicep-functions-scope#managementgroup)

_managementGroup() can only be used on a [management group deployments](https://learn.microsoft.com/en-us/azure/azure-resource-manager/bicep/deploy-to-management-group). It returns the current management group for the deployment operation. Use when either getting a scope object or getting properties for the current management group._

Reply:
A shame. One would think ARM would be able to detect the ManagementGroup of a given subscription